common/src/main/java/tachyon/ClientBase.java

```diff
@@ -188,4 +191,90 @@ protected synchronized InetSocketAddress getAddress() {
     return mAddress;
   }
 
+  /**
+   * The RPC to be executed in {@link #retryRPC(RpcCallable)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallable<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TException;
+  }
+
+  /**
+   * Same with {@link RpcCallable} except that this RPC call throws {@link TachyonTException} and
+   * is to be executed in {@link #retryRPC(RpcCallableThrowsTachyonTException)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallableThrowsTachyonTException<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TachyonTException when any {@link TachyonException} happens during RPC and is wrapped
+     *                           into {@link TachyonTException}
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TachyonTException, TException;
```

Comment:
I've done another pass on the RPCs, only few of them will only throw `TException` without possibility of `TachyonTException` or `ThriftIOException`, so +1 for this solution. 

For those that only throw `TException`, we can have a `try {} catch` around `retryRPC` to catch `TachyonException` and wrap into `IOException`, what do you think about?

Also, would we stop retrying if we encounter `ThriftIOException`? I think not, so we only need to keep `TachyonTException` and `TException` in our `RpcCallable#call()`.